haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,147 @@
+                """,
+                " - ".join([x.meta.get(self.date_identifier, "identifier wrong") for x in documents]),
+            )
+            output = {"documents": documents}
+            return output, "output_1"
+
+        # merge scores for both documents sorted by content and date.
+        # We use reciprocal rank fusion (RRF) to have the same range of scores for both lists.
+        scores_map: Dict = defaultdict(int)
+        document_map = {doc.id: doc for doc in documents}
+        weight = self.weight
+        for i, doc in enumerate(documents):
+            if self.method == "reciprocal_rank_fusion":
+                scores_map[doc.id] += self._calculate_rrf(rank=i) * (1 - weight)
+            elif self.method == "score":
+                score = float(0)
+                if doc.score is None:
+                    warnings.warn("The score was not provided; defaulting to 0")
+                elif doc.score < 0 or doc.score > 1:
+                    warnings.warn("The score is outside the [0,1] range; defaulting to 0")
+                else:
+                    score = doc.score
+
+                scores_map[doc.id] += score * (1 - weight)
+            else:
+                logger.error(
+                    """
+                    Param <method> seems wrong.\n
+                    You supplied: '%s'.\n
+                    It should be 'reciprocal_rank_fusion' or 'score'
+                    """,
+                    self.method,
+                )
+        for i, doc in enumerate(sorted_by_date):
+            if self.method == "reciprocal_rank_fusion":
+                scores_map[doc.id] += self._calculate_rrf(rank=i) * weight
+            elif self.method == "score":
+                scores_map[doc.id] += self._calc_recentness_score(rank=i, amount=len(sorted_by_date)) * weight
+        sorted_doc_ids = sorted(scores_map.items(), key=lambda d: d[1] if d[1] is not None else -1, reverse=True)
+
+        top_k = top_k or self.top_k or len(sorted_doc_ids)
+        docs = []
+        for idx, score in sorted_doc_ids[:top_k]:
+            doc = document_map[idx]
+            doc.score = score
+            docs.append(doc)
+
+        output = {"documents": docs}
+
+        return output, "output_1"
+
+    # pylint: disable=arguments-differ
+    def run_batch(self, queries: List[str], documents: Union[List[Document], List[List[Document]]], top_k: Optional[int] = None, batch_size: Optional[int] = None) -> Tuple[Dict, str]:  # type: ignore
+        if isinstance(documents[0], Document):
+            return self.run("", documents=documents, top_k=top_k)  # type: ignore
+        nested_docs = []
+        for docs in documents:
+            temp = self.run("", documents=docs, top_k=top_k)  # type: ignore
+            nested_docs.append(temp[0]["documents"])
+        output = {"documents": nested_docs}
+        return output, "output_1"
+
+    def _calculate_rrf(self, rank: int, k: int = 61) -> float:
```

Comment:
There's some minor redundancy here, since the `JoinDocuments` node also has a `_calculate_rrf` method that does essentially the same thing (see [here](https://github.com/deepset-ai/haystack/blob/22750d342ca3ad44481ba8704016174e954e20f3/haystack/nodes/other/join_docs.py#L140)). I don't think this is a big deal, but could be an area for improvement down the line.

Reply:
Thank you for the feedback Rob, I've implemented the suggestions 👍  Just for this one I'm not sure if it needs to be changed since the redundancy is indeed quite minor (for now) - if more components end up using _calculate_rrf method in the future, we can create a place under "nodes" where an rrf function and other things for using across haystack components can be stored, like a general "utils" folder or sth